satellite/metainfo/validation.go

```diff
@@ -225,12 +230,56 @@ func (endpoint *Endpoint) validateBucket(ctx context.Context, bucket []byte) (er
 	if len(bucket) == 0 {
 		return errs.New("bucket not specified")
 	}
-	if bytes.ContainsAny(bucket, "/") {
-		return errs.New("bucket should not contain slash")
+
+	if len(bucket) < 3 || len(bucket) > 63 {
+		return errs.New("bucket name must be at least 3 and no more than 63 characters long")
 	}
+
+	labels := bytes.Split(bucket, []byte("."))
```

Comment:
I have the feeling that we could check if the bucket name is correct with this regular expression `^([a-z0-9]+([a-z0-9\-][a-z0-9])*)+(.[a-z0-9]+([a-z0-9\-][a-z0-9])*)*$`

The above checks and the check `ipRegexp.MatchString` must be kept.

Reply:
I did a fast comparison with benchmark and if I didn't mess up something it looks that non regexp way is faster for valid bucket names. I think we will be processing more valid bucket names than invalid.
https://gist.github.com/mniewrzal/49de3af95f36e63e88fac24f565e444c
```
BenchmarkValidation/simple-size-5-8         	30000000	        51.7 ns/op	      32 B/op	       1 allocs/op
BenchmarkValidation/regexp-size-5-8         	 5000000	       252 ns/op	       5 B/op	       1 allocs/op
BenchmarkValidation/simple-size-29-8        	20000000	       100 ns/op	      32 B/op	       1 allocs/op
BenchmarkValidation/regexp-size-29-8        	 2000000	       991 ns/op	      32 B/op	       1 allocs/op
BenchmarkValidation/simple-size-59-8        	20000000	       121 ns/op	      32 B/op	       1 allocs/op
BenchmarkValidation/regexp-size-59-8        	 1000000	      1857 ns/op	      65 B/op	       1 allocs/op
PASS
ok  	storj.io/storj/satellite/metainfo	12.664s
```

Reply:
Let's add a comment to remember why a regex isn't used and link to the gist, where you could add a text file with this result.